lib/puppet/provider/gluster.rb

```diff
@@ -1,20 +1,51 @@
+module XPathHelpers
+  def value_if_text(node)
+    case node.node_type
+    when :text
+      node.value
+    else
+      node
+    end
+  end
+
+  def xpath_match(node, query)
+    REXML::XPath.match(node, query).map { |r| value_if_text(r) }
+  end
+
+  def xpath_first(node, query)
+    value_if_text(REXML::XPath.first(node, query))
+  end
+end
+
 class Puppet::Provider::Gluster < Puppet::Provider
 
   # NOTE: This assumes that child providers define a `gluster` command:
   #
   # has_command(:gluster, 'gluster') do
-  #   environment :HOME => "/tmp"
+  #   environment :HOME => '/tmp'
   # end
 
+  # We want these to be both class and instance methods.
+  extend XPathHelpers
+  include XPathHelpers
+
   # Peer information
 
   def self.parse_peer_status(output)
     doc = REXML::Document.new(output)
-    xpath_match(doc, "/cliOutput/peerStatus/peer/hostname/text()")
+    xpath_match(doc, '/cliOutput/peerStatus/peer/hostname/text()')
+  end
+
+  def self.gluster_cmd(*args)
+    gluster('--xml', '--mode=script', *args)
```

Comment:
That tells gluster to not ask a human for confirmation on certain actions. Since everything accepts it, it makes sense to pass it everywhere and not worry about which commands specifically need it.